add docs for auth custom attribute support (#7846)
aws-amplify/docs: src/pages/[platform]/build-a-backend/auth/concepts/user-attributes/index.mdx

```ts title="amplify/auth/resource.ts"
import { defineAuth } from "@aws-amplify/backend"

export const auth = defineAuth({
  loginWith: {
    // this configures a required "email" attribute
    email: true,
  },
  userAttributes: {
    // highlight-start
    "custom:display_name": {
      dataType: "String",
      mutable: true,
      maxLen: 16,
      minLen: 1,
    },
    "custom:favorite_number": {
      dataType: "Number",
      mutable: true,
      min: 1,
      max: 100,
    },
    "custom:is_beta_user": {
      dataType: "Boolean",
      mutable: true,
    },
    "custom:started_free_trial": {
      dataType: "DateTime",
      mutable: true,
    },
    // highlight-end
  },
})
```

Unlike standard attributes, custom attributes cannot natively be required for sign-up, however can be codified to require some value by [validating user attributes upon sign-up with a pre sign-up trigger](/[platform]/build-a-backend/functions/user-attribute-validation/).

> Custom attributes can also be configured with specific data types. The following data types are supported:

Reviewer: might be good to also list out max value range we can set or link to the doc that provides thus

```
- `String` : upto 2048 bytes
- `Number` : 2^1023
- `Boolean`
- `DateTime`
```

Reviewer: ya i think it would be good, since it might make it easier to find

- `String`
- `Number`
- `Boolean`
- `DateTime`

> Shown in the snippet above, `String` and `Number` can be assigned minimum and maximum constraints. This is useful to defer validations to the underlying service, although does not extend to validating against specific constraints such as matching against a regular expression or a range of values.

Author: ```suggestion
Shown in the snippet above, `String` and `Number` can be assigned minimum and maximum constraints. This is useful to defer simple validations to the underlying service, although does not extend to complex validations such as matching against a regular expression.
```
